### ISSUE TITLE
`Reset Game` option missing from "Tab" Menu

### ISSUE THREAD

**Issue by SapphireTactician** [libretro/mame2003-plus-libretro#136]
This seems to be relatively new. Perhaps it got accidentally removed when `Return to Game` was removed.

![2018-04-24--170031_1600x900](https://user-images.githubusercontent.com/24427487/39214030-2ce9bea4-47e1-11e8-9bc1-6573e0eccfb3.png)

**Comment by dankcushions** [comment]
retroarch already has a reset game function, though?

**Comment by SapphireTactician** [comment]
True, I'm just use to it being there so I don't have to navigate additional menus.

**Comment by dankcushions** [comment]
I'm afraid the real bug is that we have to go into the tab menu at all ;) Ultimately everything moves to the retroarch menus.

**Comment by markwkidd** [comment]
Yep you can reset from the RetroArch Quick Menu as well as the RetroArch
hotkey. That's exactly equivalent to the native MAME functionality (and
hooks into the same code). Sorry!

**Comment by dankcushions** [comment]
by default reset is hotkey + retropad B

**Comment by markwkidd** [comment]
As dankcushions says, @SapphireTactician that is a hotkey or quick menu scenario in libretro land.

This is totally lame, but it's a RetroArch issue if the Quick Menu needs to be more convenient for resetting a game.

By moving this core more into the libretro framework, it theoretically means that the Quick Menu could improve quite a bit in the future and this core will be there to benefit from the advances in RetroArch.

Hypothetically speaking of which, there is in fact going to be a new, "traditional windowed" (WIMP) menu for RetroArch starting in 1.7.3 -- it may already even make the Quick Menu even more practical on desktop systems.

**Comment by SapphireTactician** [comment]
I see. I heard about the WIMP. Personally, I'm a huge fan of fullscreen XMB.

**Comment by dankcushions** [comment]
@grant2258 if you look at the fbalpha core they do all the dips and inputs preferences via the core options. it works pretty well. i think that’s in the distant future for mame2003/plus, though.

**Comment by grant2258** [comment]
@dankcushions never used fba alpha to be honest I will have a look though. So basically what we are saying is is make things only configurable through retroarch quick menu whick is fair enough if it can be done. 

Do any other mame versions change the code as much as this or is this something new we are doing? It doesnt really matter what way we do it but if we are going to use retroarch in the distant future I guess we should re write the input  and remove mameui it in stages.

Ive never seen any port of mame do this seems odd behavior to remove a working system when all you really have to do i change the default key and joystick bindings and call the tab menu from the quick menu if you dont want the tab key to work. Unless im miss understanding something

**Comment by dankcushions** [comment]
yeah i'm not suggesting we do this now as it's so much work for basically no user benefit, but ultimately everything you can do in the tab menu you could recreate via retroarch options/input configs.

> Ive never seen any port of mame do this seems odd behavior to remove a working system when all you really have to do i change the default key and joystick bindings and call the tab menu from the quick menu if you dont want the tab key to work. Unless im miss understanding something

you could kinda say the same about making a libretro version of mame at all :)

**Comment by grant2258** [comment]
@dankcushions whatever we end up with we end up with I think marks done a good bridge to make either choice possible. I get the distinct feeling mame2003 is setup towards retropie usage I could be wrong. I hope mame2003+ can be seen as being used by a wider group of user via retroarch and reptropie.

**Comment by dankcushions** [comment]
mame2003 is not setup for retropie. what makes you say that?

**Comment by grant2258** [comment]
every discussion we have on input comes down to retropie user will have to change there input which is a fair argument. I just dont want this core in the same situation when it doesnt have to be on the input system

**Comment by markwkidd** [comment]
To be fair there is definitely a concern on the libretro side about any changes to mame2003 that would disrupt existing retropie setups. Which is fair because that is a lot of users! :grin: 

When RetroArch 1.7.2 comes to RetroPie, then people will be pretty disrupted overall and will be setting up their core options again. So it feels ok to ask them to select something like `simultaneous` if they just want to preserve their current input experience while trying out the updated core.

**Comment by twinaphex** [comment]
A libretro core should never, ever cater to any particular launcher's way of doing things. That breaks the entire libretro model of doing things.

Your frontend and your launcher should have zero bearing on decisions you make in the core.

**Comment by dankcushions** [comment]
inputs changing isn't just a concern of retropie, but a concern of any user that uses mame2003 - recalbox, retropie, retrorangepi and so on - mame2003 is a very popular core. it's not 'set up' for retropie at all. we just have to be mindful about what we change.

no-one is putting their foot down saying the inputs can't be changed in mame2003, just that we have to be careful to get it right if we're going to change it. arcade layouts are a contentious topic across libretro cores (FBA had the same issues recently) as there's no a universally accept arcade layout for a retropad.

i'm happy to have the discussion again and i'm happy to use mame2003 as the flagship for an agreed layout. i have actually previously tried to do that! https://github.com/libretro/mame2003-libretro/pull/123#issuecomment-334437679

(there's a much longer discussion in the FBA repo somewhere)

**Comment by markwkidd** [comment]
I think we can try to keep things like simultaneous mode when possible for
backwards support, and then focus on good retropad and let legacy modes
that will attract people to their benefits.

It's been like this a little with eliminating the mame hotkeys that
duplicate libretro functionality that's done in the front end. RetroPie
users are getting used to it and not complaining when it comes up in the
forums

**Comment by grant2258** [comment]
@markwkidd i think what we need is a list of retroarch hotkeys  on the wiki

**Comment by markwkidd** [comment]
I think the RetroPie docs should link to the RetroArch hotkey docs: https://docs.libretro.com/guides/retroarch-keyboard-controls/

**Comment by grant2258** [comment]
@markwkidd agreed would make more sense to do it this way. If it works properly on retroarch its done right.

**Comment by grant2258** [comment]
@Wilstorm the beautiful thing is you should be able to configure the retropad how you want if you dont like the default mapping

**Comment by markwkidd** [comment]
Minor point: This makes it easier to play some arcade games with a
keyboard-based arcade controller, and then switch to a joypad for other
games. Maybe some people want that. :shrug:
